components/AppNav.tsx

```diff
@@ -1,69 +1,64 @@
 import * as React from 'react'
 import Link from 'next/link'
+import { useRouter } from 'next/router'
 
 import '../scss/navbar.scss'
 import Button from './Button'
 
-type AuthButtonProps = {
-  initial: string
-  username: string
+import { AuthUserContext } from '../pages/index'
+
+type LinkProps = {
+  active?: string
 }
 
-type Props = {
-  loggedIn?: boolean
-  username?: string
-  firstName?: string
-  lastName?: string
+type AuthButtonProps = {
+  initial?: string
+  username: string
 }
 
-const AuthLink = () => (
+export const AuthLinks: React.FC<LinkProps> = ({ active }) => (
   <div className="navbar-nav collapse navbar-collapse">
-    <a href="#" className="nav-item nav-link active">
-      Curriculum <span className="sr-only">(current)</span>
-    </a>
-    <a href="#" className="nav-item nav-link">
+    <Link href="/curriculum">
+      <a
+        className={`nav-item nav-link${
+          active === '/curriculum' ? ' active' : ''
+        }`}
+      >
+        Curriculum <span className="sr-only">(current)</span>
+      </a>
+    </Link>
+    <a href="#" className={`nav-item nav-link`}>
       Repo
     </a>
-    <a href="#" className="nav-item nav-link">
+    <a href="#" className={`nav-item nav-link`}>
       Journey
     </a>
-    <a href="https://chat.c0d3.com" className="nav-item nav-link">
+    <a
+      href="https://chat.c0d3.com"
+      className="nav-item nav-link"
+      target="_blank"
+      rel="noopener noreferrer"
+    >
       Help
     </a>
   </div>
 )
 
-const AuthButton: React.FC<AuthButtonProps> = ({ initial, username }) => (
-  <div>
-    <Button
-      btnType="border btn-secondary overflow-hidden p-2 text-truncate"
-      initial={initial}
-      text={username}
-    />
-    <Button text="Logout" btnType="border btn-secondary ml-2" />
-  </div>
-)
-
-const UnAuthButton = () => (
-  <div>
-    <Link href="/login">
-      <a className="btn btn-secondary border mr-3">Login</a>
-    </Link>
-    <Link href="/signup">
-      <a className="btn btn-secondary border mr-3">Signup</a>
-    </Link>
-  </div>
-)
-
-const UnAuthLink = () => (
+export const UnAuthLinks: React.FC<LinkProps> = ({ active }) => (
   <div className="navbar-nav collapse navbar-collapse">
     <Link href="/">
```

Comment:
should create a new component instead of sprinkling logic: `<NavLink path="/" activePath={activePath} />`